app/stores/CollectionsStore.js

```diff
@@ -88,6 +91,28 @@ export default class CollectionsStore extends BaseStore<Collection> {
     });
   }
 
+  searchResults(query: string): CollectionSearchResult[] {
+    return this.searchCache.get(query) || [];
+  }
+
+  @action
+  search = async (query: string): Promise<CollectionSearchResult[]> => {
+    const results: CollectionSearchResult[] = compact(
+      this.orderedData.map((collection) => {
+        if (!collection.name.toLowerCase().includes(query.toLowerCase())) {
+          return null;
+        }
+
+        return {
+          collection,
+        };
+      })
+    );
+    this.searchCache.set(query, results);
```

Comment:
Let's just return the results directly from the method? The cache makes sense when searching documents because it's done async against the server – when we're just searching the local store it's unnecessary indirection.

Reply:
Understood